### PR TITLE
DOP-4829: Remove watchdog, and enable LSP file watching

### PR DESCRIPTION
**Pull request by i80and**
The existing file watching logic is quite frankly poorly designed. But Language Server Protocol
clients provide file watching, and can notify the server on file changes.

Remove our own file watching and listen to LSP file events.

In the process of this change, some missing state updates were added, and I played around a bunch with the extension. In general the extension needs more battle testing, but this is a good step forward.

We should update the extension to also monitor image files.

### README updates

- - [ ] This PR introduces changes that should be reflected in the README.md and/or HACKING.md, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README.md and/or HACKING.md

### CODE REVIEW

File: snooty/test_project.py [hunk 1]
```diff
@@ -266,25 +230,23 @@ def test_recursive_source_constants() -> None:
...
-            FileId("index.txt"),
-            ".. _gooblygooblygoo:\n",
-        )
-        project.postprocess()
-        with project._lock:
-            assert not project._project.targets["std:label:global-writes-zones"]
-            query_result = project._project.targets["std:label:gooblygooblygoo"][0]
-            assert isinstance(query_result, TargetDatabase.InternalResult)
-            assert query_result.result[0] == "index"
+    project = Project(Path("test_data/test_postprocessor"), backend, {}, watch=False)
```
Comment (rayangler) [inline snooty/test_project.py:233]: Looks like `watch` is no longer used for anything in Project. Should we keep the field defined (both here and within the class init)?

**Review by rayangler — COMMENTED** [review]
I noticed we have a [stub](https://github.com/mongodb/snooty-parser/tree/main/stubs/watchdog) for watchdogs. Is that something we should still keep?

File: snooty/language_server.py [hunk 2]
```diff
@@ -541,16 +584,25 @@ def m_text_document__did_close(self, textDocument: SerializableType) -> None:
...
+            logger.info("did_change_watched_files: " + str(fileid))
+
+            self.update_file(fileid, FileChangeType(change.type))
 
     def m_shutdown(self, **_kwargs: object) -> None:
         self._shutdown = True
 
     def m_exit(self, **_kwargs: object) -> None:
         self._endpoint.shutdown()
         self._debouncer.stop()
```
[the remark this answers is not in the document]
Comment (i80and) [reply snooty/language_server.py:605]: Why yes we should! Wonderful catch!

File: snooty/test_project.py (hunk 1 above)
Comment (i80and) [reply snooty/test_project.py]: Indeed! It is now unused

**Review by rayangler — APPROVED** [review]
Code changes LGTM! General testing with the extension looked fine based on what I can tell.

**Pull request merged**